rskj-core/src/main/java/org/ethereum/listener/CompositeEthereumListener.java

```diff
@@ -52,113 +62,94 @@ public void removeListener(EthereumListener listener) {
-        for (EthereumListener listener : listeners) {
-            listener.onTransactionPoolChanged(transactionPool);
-        }
+        scheduleListenerCallbacks(listener -> listener.onTransactionPoolChanged(transactionPool));
     }
 
     @Override
     public void onSyncDone() {
-        for (EthereumListener listener : listeners) {
-            listener.onSyncDone();
-        }
+        scheduleListenerCallbacks(EthereumListener::onSyncDone);
     }
 
     @Override
     public void onNoConnections() {
-        for (EthereumListener listener : listeners) {
-            listener.onNoConnections();
-        }
+        scheduleListenerCallbacks(EthereumListener::onNoConnections);
     }
 
     @Override
     public void onHandShakePeer(Channel channel, HelloMessage helloMessage) {
-        for (EthereumListener listener : listeners) {
-            listener.onHandShakePeer(channel, helloMessage);
-        }
+        scheduleListenerCallbacks(listener -> listener.onHandShakePeer(channel, helloMessage));
     }
 
     @Override
     public void onVMTraceCreated(String transactionHash, ProgramTrace trace) {
-        for (EthereumListener listener : listeners) {
-            listener.onVMTraceCreated(transactionHash, trace);
-        }
+        scheduleListenerCallbacks(listener -> listener.onVMTraceCreated(transactionHash, trace));
     }
 
     @Override
     public void onNodeDiscovered(Node node) {
-        for (EthereumListener listener : listeners) {
-            listener.onNodeDiscovered(node);
-        }
+        scheduleListenerCallbacks(listener -> listener.onNodeDiscovered(node));
     }
 
     @Override
     public void onEthStatusUpdated(Channel channel, StatusMessage status) {
-        for (EthereumListener listener : listeners) {
-            listener.onEthStatusUpdated(channel, status);
-        }
+        scheduleListenerCallbacks(listener -> listener.onEthStatusUpdated(channel, status));
     }
 
     @Override
     public void onTransactionExecuted(TransactionExecutionSummary summary) {
-        for (EthereumListener listener : listeners) {
-            listener.onTransactionExecuted(summary);
-        }
+        scheduleListenerCallbacks(listener -> listener.onTransactionExecuted(summary));
     }
 
     @Override
     public void onPeerAddedToSyncPool(Channel peer) {
-        for (EthereumListener listener : listeners) {
-            listener.onPeerAddedToSyncPool(peer);
-        }
+        scheduleListenerCallbacks(listener -> listener.onPeerAddedToSyncPool(peer));
     }
 
     @Override
     public void onLongSyncDone() {
-        for (EthereumListener listener : listeners) {
-            listener.onLongSyncDone();
-        }
+        scheduleListenerCallbacks(EthereumListener::onLongSyncDone);
     }
 
     @Override
     public void onLongSyncStarted() {
+        scheduleListenerCallbacks(EthereumListener::onLongSyncStarted);
+    }
+
+    private void scheduleListenerCallbacks(Consumer<EthereumListener> callback) {
```

Comment:
I like this function, maybe this is a good moment to think if we want this stuff to be a panic

Reply:
I think it's a good idea to panic, because listeners should be handling exceptions